### PR TITLE
call shutdown, if a panic is triggered

### PR DESCRIPTION
**Pull request by stlankes**
This patch should solve #45. @jounathaen Can you test it also on your system?

### CODE REVIEW

**Comment by jounathaen** [comment]
I think, this solves #45 but if executed on "Real Hardware" would result in the Device to shut down and the user not beeing able to read the panic message

File: src/arch/x86_64/kernel/processor.rs [hunk 1]
```diff
@@ -288,6 +290,22 @@ impl CpuFrequency {
 		}
 	}
 
+	unsafe fn detect_from_cpuid_hypervisor_info(&mut self, cpuid: &CpuId) -> Result<(), ()> {
+		match cpuid.get_hypervisor_info() {
```
Comment (jounathaen) [inline src/arch/x86_64/kernel/processor.rs:294]: This is an `if let Some(...)` pattern

Comment (stlankes) [reply src/arch/x86_64/kernel/processor.rs]: Done

**Comment by ColinFinck** [comment]
FYI, I recently discovered an even nicer alternative to such nested `if let` statements - one that doesn't need any nesting at all:

```
let tsc_info = cpuid.get_tsc_info().ok_or(())?;
let freq = tsc_info.tsc_frequency().ok_or(())?;
let mhz = (freq / 1000000u64) as u16;
self.set_detected_cpu_frequency(mhz, CpuFrequencySources::CpuIdTscInfo)
```

**Pull request merged**